All_home_works/hw8/task1.py

```diff
@@ -0,0 +1,70 @@
+"""
+We have a file that works as key-value storage,
+each line is represented as key and value separated by = symbol, example:
+
+name=kek
+last_name=top
+song_name=shadilay
+power=9001
+
+Values can be strings or integer numbers.
+If a value can be treated both as a number and a string, it is treated as number.
+
+Write a wrapper class for this key value storage that works like this:
+
+storage = KeyValueStorage('path_to_file.txt')
+that has its keys and values accessible as collection items and as attributes.
+Example:
+storage['name'] # will be string 'kek'
+storage.song_name # will be 'shadilay'
+storage.power # will be integer 9001
+
+In case of attribute clash existing built-in attributes take precedence.
+In case when value cannot be assigned to an attribute
+(for example when there's a line 1=something) ValueError should be raised.
+File size is expected to be small, you are permitted to read it entirely into memory.
+"""
+import string
+
+
+class KeyValueStorage:
+    allowed_chars = set(string.digits) | set(string.ascii_letters) | set("_")
+
+    def __init__(self, path):
+        self.path = path
+        self.data = self._get_data_from_file()
+
+    def _get_lines_from_file(self):
+        with open(self.path) as file:
+            for line in file:
+                yield line.rstrip("\n")
+
+    def _get_data_from_file(self):
+        data = {}
+        for line in self._get_lines_from_file():
+            attr_name, attr_value = line.split("=")
+            attr_value = int(attr_value) if attr_value.isdigit() else attr_value
+            self._check_attr_name_on_syntax(attr_name)
+            data[attr_name] = attr_value
+        return data
+
+    def _check_attr_name_on_syntax(self, name):
```

Comment:
Please check the batteries.

Reply:
Are you talking about naming, because I don't understand batteries=)?

Reply:
This is a reference to "batteries included".

Reply:
Thanks! I Will remember it `str.isidentifier`.